src/pykeen/models/base.py

```diff
@@ -563,6 +563,27 @@ def predict_scores_all_relations(
             scores = torch.sigmoid(scores)
         return scores
 
+    def _score_with_inverse_relations(
+        self,
+        hrt_batch: torch.LongTensor,
+    ) -> torch.FloatTensor:
+        """Score triples based on inverse triples, i.e., compute f(h,r,t) based on f(t,r_inv,h)."""
+        if not self.triples_factory.create_inverse_triples:
+            raise Exception("Model is not configured to predict with inverse relations. "
+                            "You might set self.triples_factory.create_inverse_triples=True when creating"
+                            "the triples the factory.")
+        hrt_batch_cloned = hrt_batch.clone()
+        hrt_batch_cloned.to(device=hrt_batch.device)
```

Comment:
clone keeps the device

Reply:
> @mali-git this PR introduces a lot of code duplication.
> 
> I think we could improve by adding another separate method, e.g. `score_hrt_inverse` which then does what `_score_with_inverse_relations` does. Hence, there would be no need to change any base classes.

Good point! I addressed this in cd2580014fddafc75b5a91154e5ec9968536a8a7, 03bc8bb790adf918a6691f524373190b10822c50, and 8fbbc660958bc6c3dc76fa56f84a16960fc14be7.